src/iotconnect_discovery.c

```diff
@@ -119,46 +133,59 @@ IotclSyncResponse *iotcl_discovery_parse_sync_response(const char *response_data
         response->ds = IOTCL_SR_PARSING_ERROR;
         return response;
     }
-    tmp_value = cJSON_GetObjectItem(sync_res_json, "ds");
-    if (!tmp_value) {
+
+    if (response->ds == IOTCL_SR_PARSING_ERROR) {
         response->ds = IOTCL_SR_PARSING_ERROR;
     } else {
-        response->ds = cJSON_GetNumberValue(tmp_value);
+        response->ds = IOTCL_SR_OK;
     }
+
     if (response->ds == IOTCL_SR_OK) {
-        response->cpid = safe_get_string_and_strdup(sync_res_json, "cpId");
-        response->dtg = safe_get_string_and_strdup(sync_res_json, "dtg");
-        tmp_value = cJSON_GetObjectItem(sync_res_json, "ee");
-        if (!tmp_value) {
-            response->ee = -1;
-        }
-        tmp_value = cJSON_GetObjectItem(sync_res_json, "rc");
-        if (!tmp_value) {
-            response->rc = -1;
-        }
-        tmp_value = cJSON_GetObjectItem(sync_res_json, "at");
-        if (!tmp_value) {
-            response->at = -1;
-        }
+        cJSON *meta_json = cJSON_GetObjectItemCaseSensitive(sync_res_json, "meta");
+		if (!meta_json) {
+		    cJSON_Delete(sync_json_root);
+		    response->ds = IOTCL_SR_PARSING_ERROR;
+		    return response;
+		}
+
+		response->cd = safe_get_string_and_strdup(meta_json, "cd");
+
+		if (!response->cd) {
+            cJSON_Delete(sync_json_root);
+            response->ds = IOTCL_SR_PARSING_ERROR;
+            return response;
+		}
+
         cJSON *p = cJSON_GetObjectItemCaseSensitive(sync_res_json, "p");
+
         if (p) {
+
             response->broker.name = safe_get_string_and_strdup(p, "n");
             response->broker.client_id = safe_get_string_and_strdup(p, "id");
             response->broker.host = safe_get_string_and_strdup(p, "h");
             response->broker.user_name = safe_get_string_and_strdup(p, "un");
+            response->broker.port = get_numeric_value_or_default(p, "p", 8883);
+
+#if 0
```

Comment:
Is there a particular reason to hide this behind an "if 0"?

Does the old code need to stay visible or be available for reversion later?

In any case, I'd prefer to have a comment explaining that fact and/or use a dsecriptive preproscessor define

(e.g. #if DEPRECATED_PRE_AWS or something)